dask_match/tests/test_collection.py

```diff
@@ -139,12 +139,38 @@ def test_repr(ddf):
 
 
 def test_columns_traverse_filters(df, ddf):
-    result = optimize(ddf[ddf.x > 5].y)
+    result = optimize(ddf[ddf.x > 5].y, fuse=False)
     expected = ddf.y[ddf.x > 5]
 
     assert str(result) == str(expected)
 
 
+def test_optimize_fusion(ddf):
+
+    ddf2 = (ddf["x"] + ddf["y"]) - 1
+    unfused = optimize(ddf2, fuse=False)
+    fused = optimize(ddf2, fuse=True)
+
+    # Should only get one task per partition
+    assert len(fused.dask) == ddf.npartitions
+    assert_eq(fused, unfused)
+
+    # Check that we still get fusion when
+    # non-blockwise operations are added
+    ddf3 = ddf2.sum()
+    unfused = optimize(ddf3, fuse=False)
+    fused = optimize(ddf3, fuse=True)
+
+    assert len(fused.dask) < len(unfused.dask)
+    assert_eq(fused, unfused)
+
+    # Check that we still get fusion
+    # after a non-blockwise operation as well
+    fused_2 = optimize(ddf3 + 10 - 5, fuse=True)
```

Comment:
maybe explicitly parenthesise `(ddf3 + 10) - 5` so people don't have to remember associativity rules.